pkg/controller/stats/aggregator.go

```diff
@@ -369,3 +371,36 @@ func addUp(stats *statsv1alpha1.TrafficStats, inc *statsv1alpha1.TrafficStats) {
 	stats.Packets += inc.Packets
 	stats.Bytes += inc.Bytes
 }
+
+func addRulesUp(ruleStats *[]statsv1alpha1.RuleTrafficStats, ruleSumStats *statsv1alpha1.TrafficStats, inc map[string]statsv1alpha1.TrafficStats) {
+	// accumulate inc traffics stats to the current traffic stats
+	for _, v := range inc {
+		ruleSumStats.Bytes += v.Bytes
+		ruleSumStats.Sessions += v.Sessions
+		ruleSumStats.Packets += v.Packets
+	}
+	// if the a rule of current traffic stats in the inc map, sum the traffic stats to this rule and delete the rule from inc
+	for i, v := range *ruleStats {
+		stats, exist := inc[v.Name]
+		if exist {
+			(*ruleStats)[i].TrafficStats = statsv1alpha1.TrafficStats{
+				Packets:  v.TrafficStats.Packets + stats.Packets,
+				Bytes:    v.TrafficStats.Bytes + stats.Bytes,
+				Sessions: v.TrafficStats.Sessions + stats.Sessions,
+			}
+		}
+		delete(inc, v.Name)
+	}
+	// convert remaining incs to ruletrafficstats and add it to current traffic stats
```

Comment:
If you iterate inc in L383 instead, it can just update current stats if it exists or add new stats if not.

Reply:
If iterate `inc`, I have to traverse the slice `ruleStats` to match the rule name in each iteration, which makes the implementation O^2

Reply:
I see, didn't notice ruleStats is a slice. Thanks for the explanation.